marin/export/hf_upload.py

```diff
@@ -0,0 +1,259 @@
+                commit_message += f"- {path_in_repo}\n"
+
+                if batch_bytes > max_size_bytes:
+                    safe_create_commit(
+                        config.repo_id,
+                        operations=batch,
+                        commit_message=base_message,
+                        token=config.token,
+                        commit_description=commit_message,
+                        repo_type=config.repo_type,
+                        revision=config.revision,
+                    )
+                    pbar.update(batch_bytes)
+                    batch = []
+                    batch_bytes = 0
+                    commit_message = base_message
+
+            if batch:
+                safe_create_commit(
+                    config.repo_id,
+                    operations=batch,
+                    commit_message=base_message,
+                    token=config.token,
+                    commit_description=commit_message,
+                    repo_type=config.repo_type,
+                    revision=config.revision,
+                )
+                pbar.update(batch_bytes)
+
+        # Upload the small files using upload_folder
+        if tiny_files:
+            logger.info(f"Uploading {len(tiny_files)} small files to {config.repo_id}")
+            with tempfile.TemporaryDirectory() as tmpdir:
+                for path in tiny_files:
+                    path_in_repo = os.path.relpath(path, config.input_path)
+                    fs.get(path, os.path.join(tmpdir, path_in_repo))
+
+                safe_upload_folder(
+                    folder_path=tmpdir,
+                    repo_id=config.repo_id,
+                    repo_type=config.repo_type,
+                    token=config.token,
+                    revision=config.revision,
+                    commit_message=f"Uploading small files from {config.input_path}",
+                    **config.upload_kwargs,
+                )
+
+
+def with_retries(max_retries=3, delay=2.0):
+    def decorator(fn):
+        @wraps(fn)
+        def wrapped(*args, **kwargs):
+            for attempt in range(max_retries):
+                try:
+                    return fn(*args, **kwargs)
+                except Exception as e:
+                    if attempt == max_retries - 1:
+                        raise
+                    logger.warning(f"{fn.__name__} failed on attempt {attempt + 1}, retrying: {e}")
+                    time.sleep(delay)
+
+            raise RuntimeError(f"{fn.__name__} failed after {max_retries} attempts")
+
+        return wrapped
+
+    return decorator
+
+
+@functools.wraps(upload_folder)
+@with_retries()
+def safe_upload_folder(*args, **kwargs):
+    return upload_folder(*args, **kwargs)
+
+
+@functools.wraps(create_commit)
+@with_retries()
+def safe_create_commit(*args, **kwargs):
+    return create_commit(*args, **kwargs)
+
+
+if __name__ == "__main__":
+    # dummy test
+
+    with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
Is it actually possible to have tests?

Reply:
it'd be a real pain to mock out all of HF's stuff so I don't think it's worth it